cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/AbstractProgramWorkflowRunner.java

```diff
@@ -136,6 +137,13 @@ public void error(Throwable cause) {
         throw (Exception) cause;
       }
       throw Throwables.propagate(cause);
+    } catch (InterruptedException e) {
+      try {
+        Futures.getUnchecked(controller.stop());
+      } catch (Throwable t) {
+        // no-op
+      }
+      throw e;
```

Comment:
If you changed the state of the controller (by calling stop), you shouldn't need to propagate the interruptedException, since the caller should simply be watching the state changes in the controller to determine what state the program is in.